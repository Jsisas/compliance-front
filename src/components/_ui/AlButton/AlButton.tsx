import React, { ReactNode } from 'react';
import { Button } from 'antd';
import styles from './alButton.module.scss';
import { CSSProperties } from 'styled-components';
import { concatStyles } from '../../../util/StyleUtil';

export interface ButtonProp {
	type: 'primary' | 'secondary' | 'link';
	style?: CSSProperties;
	children?: ReactNode;
	onClick?: (event: React.MouseEvent<HTMLElement, MouseEvent>) => void;
	htmlType?: 'button' | 'submit' | 'reset';
	className?: string | '';
	disabled?: boolean;
	preventDefault?: boolean;
	stopPropagation?: boolean;
}

export default function AlButton(props: ButtonProp): JSX.Element {
	function onButtonClick(e: React.MouseEvent<HTMLElement, MouseEvent>): void {
		if (props.preventDefault) {
			e.preventDefault();
		}

		if (props.stopPropagation) {
			e.stopPropagation();
		}

		if (props.onClick) {
			props.onClick(e);
		}
	}

	function getButtonClassName(): string {
		if (props.type === 'link') {
			return '';
		} else {
			return styles[props.type];
		}
	}

	function getButtonType(type: string): 'link' | 'primary' | undefined {
		if (type === 'link' || type === 'primary') {
			return type;
		} else if (type === 'secondary') {
			return undefined;
		} else {
			return 'primary';
		}
	}

	return (
		<Button
			className={concatStyles(getButtonClassName(), props.className || '')}
			type={getButtonType(props.type)}
			style={props.style}
			onClick={onButtonClick}
			htmlType={props.htmlType}
			disabled={props.disabled || false}
		>
			{props.children}
		</Button>
	);
}
