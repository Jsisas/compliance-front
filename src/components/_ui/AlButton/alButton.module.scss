@import "../../../theme.module";

.primary {
  border: none;
  border-radius: 4px;
  background: $success-color;

  &:hover, &:focus {
    background: $success-color-hover;
    border-color: unset;
  }

  &:not(:hover){
    background: $success-color;
  }
}

.primary-disabled {
  background: $success-color;
}

.secondary {
  border-radius: 4px;
  background: $white;
  border: 1px solid $light-grey;
  font-weight: 600;
  font-size: 15px;
  line-height: 20px;
  color: $text-color;

  &:hover, &:focus {
    color: $text-color;
    border-color: $dark-grey;
    background: $white;
  }

  &:not(:hover){
    border-color: $light-grey;
  }
}
