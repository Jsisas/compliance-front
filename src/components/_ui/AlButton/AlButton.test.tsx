import React from 'react';
import { shallow } from 'enzyme';
import AlButton from './AlButton';

describe('<AlButton />', () => {
	it('it renders a button', () => {
		const wrapper = shallow(<AlButton type={'primary'} />);
		expect(wrapper.exists('Button')).toBe(true);
		expect(wrapper).toMatchSnapshot();
	});

	it('primary button attributes are correct', () => {
		const onClickMock = jest.fn();
		const wrapper = shallow(<AlButton type={'primary'} />);

		wrapper.setProps({
			type: 'primary',
			style: { position: 'absolute' },
			onClick: onClickMock,
			htmlType: 'button',
			className: 'test-class',
			disabled: true,
		});

		const antdButton = wrapper.find('Button');
		expect(antdButton.prop('className')).toBe('primary test-class');
		expect(antdButton.prop('type')).toBe('primary');
		expect(antdButton.prop('style')).toStrictEqual({ position: 'absolute' });
		expect(antdButton.prop('htmlType')).toBe('button');
		expect(antdButton.prop('disabled')).toBe(true);

		antdButton.simulate('click');
		expect(onClickMock).toHaveBeenCalled();
	});

	it('link button attributes are correct', () => {
		const wrapper = shallow(<AlButton type={'link'} />);

		wrapper.setProps({
			type: 'link',
		});

		const antdButton = wrapper.find('Button');
		expect(antdButton.prop('type')).toBe('link');
		expect(antdButton.prop('className')?.trim()).toBe('');
	});

	it('secondary button attributes are correct', () => {
		const wrapper = shallow(<AlButton type={'secondary'} />);

		wrapper.setProps({
			type: 'secondary',
		});

		const antdButton = wrapper.find('Button');
		expect(antdButton.prop('type')).toBe(undefined);
		expect(antdButton.prop('className')?.trim()).toBe('secondary');
	});
});
