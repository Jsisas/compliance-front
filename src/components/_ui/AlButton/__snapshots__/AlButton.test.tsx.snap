// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlButton /> it renders a button 1`] = `ShallowWrapper {}`;
